core/src/test/scala/org/apache/spark/storage/CsdUnrollSafetySuite.scala

```diff
@@ -0,0 +1,282 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.storage
+
+import org.apache.spark._
+import org.apache.spark.memory.StaticMemoryManager
+import org.apache.spark.network.netty.NettyBlockTransferService
+import org.apache.spark.rpc.RpcEnv
+import org.apache.spark.scheduler.LiveListenerBus
+import org.apache.spark.serializer.KryoSerializer
+import org.apache.spark.shuffle.hash.HashShuffleManager
+import org.apache.spark.util._
+import org.scalatest._
+
+import scala.collection.mutable.ArrayBuffer
+import scala.language.{implicitConversions, postfixOps}
+
+/**
+ * This suite covers [[org.apache.spark.storage.MemoryStore.unrollSafely]]
+ * for CSD's cache black size policy
+ */
+class CsdUnrollSafetySuite extends SparkFunSuite with Matchers with BeforeAndAfterEach
```

Comment:
The relevant method if named `unrollSafely`, so this class and file should be named `CsdUnrollSafelySuite`.

Reply:
Actually, since almost everything except the tests themselves is copied from `BlockManagerSuite`, can't these new tests just be added to that suite?

Reply:
We do have an extra con setup in beforeEach, so that our size limit can take effect.
```
    conf.set("spark.storage.MemoryStore.csdCacheBlockSizeLimit", "8000")
```

Reply:
Yes, but adding that shouldn't break any of the existing `BlockManagerSuite` tests, right?

Reply:
yes, when we impose the block size limit, the the behavior changes and some of the tests will need to be modified to reflect the fact.
I will try this:
```
class CsdUnrollSafelySuite extends BlockManagerSuite {
```
And override the tests that assert the states we cared.

Reply:
Either that or try re-setting the conf and the `master` var within the `BlockManagerSuite` tests to run similar tests with both configurations.